Tasks/Maven/maventask.ts

```diff
@@ -258,10 +314,23 @@ mvnv.exec()
     console.error("SonarQube analysis failed");
     sqRunFailed = true;
 })
+.then(function (code) { // Pick up files from the Java code analysis tools
+    if (userRunFailed) {
+        console.error('Could not retrieve code analysis results - Maven run failed.');
```

Comment:
A test failure may cause the run to fail but you still can run Pmd. We discussed about this offline and decided we don't want to introduce more complex logic here. It is worth adding a comment in code explaining this!